utils.py

```diff
@@ -0,0 +1,332 @@
+            last_key += 1
+        except IndexError:
+            last_key = 1
+        transactions_data.setdefault(last_key, data)
+        file_path = create_file_path(
+            BASE_JSON_DIR, TRANSACTION_FILENAME_FORMAT.format(username=username)
+        )
+        write_json(file_path, transactions_data)
+
+
+def check_ticket_quantity(available_ticket: int) -> int:
+    """
+    function to check if ticket quantity for specific slot is available
+    @param available_ticket: int
+    @return: int
+    """
+    if available_ticket == 0:
+        print("Houseful")
+        exit()
+    tries = 3
+    while True:
+        ticket_quantity = user_input_int_type("Select Ticket Quantity: ")
+        if ticket_quantity <= available_ticket:
+            return ticket_quantity
+        elif tries == 0:
+            raise UnavailableTickets("Try Booking the Tickets the are Available")
+        else:
+            print(TICKET_QUANTITY_CHECK.format(tries=tries))
+            tries -= 1
+
+
+def get_date() -> date:
+    """
+    function to return date object
+    @return: date
+    """
+    d = int(input("Enter Date[1-31]: "))
+    m = int(input("Enter Month[1-12]: "))
+    y = int(input("Enter Year: "))
+    return date(y, m, d)
+
+
+def generate_slots(duration: int, seats: int) -> dict:
+    """
+    function to generate slots: seats respectively
+    @param duration: int
+    @param seats: int
+    @return:
+    """
+    duration = str(duration) + "min"
+    datetime_slots = pandas.date_range(
+        start="12:00", end="23:00", freq=duration
+    ).strftime("%H:%M")
+    return dict.fromkeys(datetime_slots, seats)
+
+
+def create_time_slots(
+    start_date: date, end_date: date, duration: int, seats: int
+) -> dict:
+    """
+    function to create time slots based on given start and end date
+    @param seats: int
+    @param duration: int
+    @param start_date: date
+    @param end_date: ate
+    @return: dict
+    """
+    slots = {}
+    slots_seats_data = generate_slots(duration, seats)
+    while start_date <= end_date:
+        slots.setdefault(str(start_date), slots_seats_data)
+        start_date = start_date + timedelta(days=1)
+    return slots
+
+
+def generate_movie_list_based_on_date(movie_date: dict) -> list:
+    """
+    function to return movie list enumerate object based on date
+    @param movie_date: dict
+    @return:list
+    """
+    upcoming_movie = []
+    past_movies = []
+    for key, value in movie_date.items():
+        if str(date.today()) <= value.get("end_date"):
+            upcoming_movie.append(key)
+        else:
+            past_movies.append(key)
+    return [upcoming_movie, past_movies]
+
+
+def get_available_date_slots(date_slots: dict) -> dict:
+    """
+    function to return date slots counter based on current date
+    @param date_slots: dict
+    @return: dict
+    """
+    counter = 1
+    date_slot_counter = {}
+    for key in date_slots.keys():
```

Comment:
key what does it mean key is predefined keyword in python , It should be meaningful name